### ISSUE TITLE
TypeError: Cannot read property 'detach' of undefined

### ISSUE THREAD

**Issue by chinabrother** [msarsha/ng2-right-click-menu#61]
Problem arises when destroyed elements with menu.

overlayRef = undefined

> src/lib/src/sh-context-menu.component.ts

```
  close(): void {
    this.setNotActive();
    this.menuContainer.detach();
    this.overlayRef.detach();
  }
```

**Comment by msarsha** [comment]
Hey !

Thank you for reporting and submitting a PR (#62)

Can you please add steps to be able to replicate the error ?

**Comment by chinabrother** [comment]
1. Open page with module (two module and more)
2. Go over href to other module with loading loadChildren

**Comment by ruisebastiao** [comment]
+1
same here

**Comment by Observablerxjs** [comment]
@msarsha are you going to publish a new version on NPM?

**Comment by msarsha** [comment]
Yes.

Hope to have some free time tomorrow morning.

**Comment by msarsha** [comment]
published v1.0.1
